drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -636,26 +637,156 @@ static int bq274xx_gauge_configure(const struct device *dev)
 		return -EIO;
 	}
 
+#ifdef CONFIG_PM_DEVICE
+	bq274xx->pm_state = PM_DEVICE_STATE_ACTIVE;
+#endif
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int bq274xx_enter_shutdown_mode(struct bq274xx_data *data)
+{
+	int status = 0;
+
+	status = bq274xx_control_reg_write(data, BQ274XX_UNSEAL_KEY);
+	if (status < 0) {
+		LOG_ERR("Unable to unseal the battery");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_UNSEAL_KEY);
+	if (status < 0) {
+		LOG_ERR("Unable to unseal the battery");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data,
+					   BQ274XX_CONTROL_SHUTDOWN_ENABLE);
+	if (status < 0) {
+		LOG_ERR("Unable to enable shutdown mode");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_CONTROL_SHUTDOWN);
+	if (status < 0) {
+		LOG_ERR("Unable to enter shutdown mode");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_CONTROL_SEALED);
+	if (status < 0) {
+		LOG_ERR("Failed to seal the gauge");
+		return -EIO;
+	}
+
+	data->pm_state = PM_DEVICE_STATE_OFF;
+
+	return 0;
+}
+
+static int bq274xx_exit_shutdown_mode(const struct device *dev)
+{
+	const struct bq274xx_config *const config = dev->config;
+	int status = 0;
+
+	if (!device_is_ready(config->gpio_pm.port)) {
+		LOG_ERR("GPIO device pointer is not ready to be used");
+		return -EINVAL;
+	}
+
+	status = gpio_pin_configure_dt(&config->gpio_pm,
+			   GPIO_OUTPUT | GPIO_OPEN_DRAIN);
+	if (status < 0) {
+		LOG_ERR("Unable to configure interrupt pin to output and open drain");
+		return status;
+	}
+
+	status = gpio_pin_set(config->gpio_pm.port, config->gpio_pm.pin, 0);
+	if (status < 0) {
+		LOG_ERR("Unable to set interrupt pin to low");
+		return status;
+	}
+
+	k_msleep(1);
+
+	status = gpio_pin_configure_dt(&config->gpio_pm,
+			   GPIO_INPUT);
+	if (status < 0) {
+		LOG_ERR("Unable to configure interrupt pin to input");
+		return status;
+	}
+
+	k_msleep(100);
+
+	status = bq274xx_gauge_configure(dev);
+	if (status < 0) {
+		LOG_ERR("Unable to configure bq274xx gauge");
+		return status;
+	}
+
 	return 0;
 }
 
+static int bq274xx_pm_control(const struct device *dev, uint32_t ctrl_command,
+				  void *context, pm_device_cb cb, void *arg)
```

Comment:
please use the updated signature:

```suggestion
				  enum pm_device_state state)
```

Some more changes are coming, but this is the current callback signature.

ref. #36756 #36757